pkg/daemon/plugin.go

```diff
@@ -71,12 +71,20 @@ func enablePlugins(platform utils.PlatformType, useSystemdService bool, ns *srio
 	return enabledPlugins, nil
 }
 
-func registerVendorPlugins(ns *sriovnetworkv1.SriovNetworkNodeState) (map[string]plugin.VendorPlugin, error) {
+func registerVendorPlugins(ns *sriovnetworkv1.SriovNetworkNodeState, storeManager utils.StoreManagerInterface) (map[string]plugin.VendorPlugin, error) {
 	vendorPlugins := map[string]plugin.VendorPlugin{}
+	vendorSet := map[string]struct{}{}
 
+	// populate Set of vendors for plugin creation
```

Comment:
while at it, i added this to instantiate plugin only once per vendor, hope its ok

Reply:
Why we need a separate set here? Is it possible to simply check if existing `vendorPlugins` map already contains a vendor name and skip plugin initialization in this case?

Reply:
going back to this change, 

vendorPlugins is keyed by plugin name and not by vendor.

to instantiate once we then need this set.
if we want just to set a plugin once in vendorPlugins map then we can do as u say.

seeing its not a really big deal, i will just set plugin in map once.